### PR TITLE
Chain / RStudio: exclude entities without data

### LINKED ISSUE #2294: Chain: an entity without data in RStudio
An entity without any data is in RStudio as a list, not data frame. It does not contain _uuid fields,

In addition, persist-results.R crashes for several reasons.

Solution: Exclude blank entities and their result attributes (both categorical and quantitative) from the chain imported into RStudio.

### PR DESCRIPTION
**Pull request by SteRiccio**
resolves #2294

### CODE REVIEW

File: server/modules/surveyRdb/api/surveyRdbApi.js [hunk 1]
```diff
@@ -52,6 +52,18 @@ export const init = (app) => {
     }
   )
 
+  app.get('/surveyRdb/:surveyId/view_data_counts', requireRecordListViewPermission, async (req, res, next) => {
```
Comment (ramirobg94) [inline server/modules/surveyRdb/api/surveyRdbApi.js:55]: the name is weird, count the number of data into a view?
or count the number of tables by entity as the code says, 

also countsByUuid -> countXXXXXXByUuid

File: server/modules/analysis/service/rChain/rChain.js [hunk 2]
```diff
@@ -136,7 +137,19 @@ class RChain {
...
+
+    const dataViewCountsByEntityDefUuid = await SurveyRdbManager.countTables({
+      survey: this.survey,
+      cycle: this.cycle,
+      entityDefUuids: entityDefs.map(NodeDef.getUuid),
+    })
+    const entitiesWithData = entityDefs.filter((entityDef) => {
+      const dataViewCount = dataViewCountsByEntityDefUuid[NodeDef.getUuid(entityDef)]
+      return dataViewCount > 0
+    })
```
[the remark this answers is not in the document]
Comment (SteRiccio) [reply server/modules/analysis/service/rChain/rChain.js:150]: ok, I'll do it

File: server/modules/surveyRdb/api/surveyRdbApi.js (hunk 1 above)
Comment (SteRiccio) [reply server/modules/surveyRdb/api/surveyRdbApi.js:55]: ah, maybe we can call it count_view_data_rows or just count_view_data?

Comment (ramirobg94) [reply server/modules/surveyRdb/api/surveyRdbApi.js]: does it count the number of rows?

Comment (SteRiccio) [reply server/modules/surveyRdb/api/surveyRdbApi.js]: Yes, it's the count of rows in each entity data view

Comment (ramirobg94) [reply server/modules/surveyRdb/api/surveyRdbApi.js]: It is quite weird,

**Review by ramirobg94 — APPROVED** [review]

**Pull request merged**